### PR TITLE
misc: Add gosec to check security issues

### PR DESCRIPTION
**Pull request by jimmyhu-dexon**

### CODE REVIEW

File: core/crypto/dkg/dkg.go [hunk 1]
```diff
@@ -276,6 +278,7 @@ func (pubs *PublicKeyShares) Clone() *PublicKeyShares {
 // NewID creates a ew ID structure.
 func NewID(id []byte) ID {
 	var blsID bls.ID
+	// #nosec G104
```
Comment (missionliao) [inline core/crypto/dkg/dkg.go]: Suggestion: personally, I would prefer to panic if error raised here.

File: core/crypto/dkg/dkg.go [hunk 2]
```diff
@@ -325,6 +329,7 @@ func (prvs *PrivateKeyShares) SetParticipants(IDs IDs) {
 	prvs.shares = make([]PrivateKey, len(IDs))
 	prvs.shareIndex = make(map[ID]int, len(IDs))
 	for idx, ID := range IDs {
+		// #nosec G104
```
Comment (missionliao) [inline core/crypto/dkg/dkg.go]: Suggestion: maybe we can just return error and break the loop.

File: core/test/fake-transport.go [hunk 3]
```diff
@@ -104,6 +104,7 @@ func (t *FakeTransport) Broadcast(endpoints map[types.NodeID]struct{},
 		}
 		go func(nID types.NodeID) {
 			time.Sleep(latency.Delay())
+			// #nosec G104
```
Comment (missionliao) [inline core/test/fake-transport.go]: Suggestion: maybe we could panic here.

File: core/test/tcp-transport.go [hunk 4]
```diff
@@ -632,6 +638,7 @@ func (t *TCPTransportClient) Join(
 			if nID == t.nID {
 				break
 			}
+			// #nosec G104
```
Comment (missionliao) [inline core/test/tcp-transport.go]: Suggestion: maybe we could panic here.

File: simulation/app.go [hunk 5]
```diff
@@ -168,6 +168,7 @@ func (a *simApp) TotalOrderingDelivered(
 		BlockHash:      blockHashes,
 		ConfirmLatency: latencies,
 	}
+	// #nosec G104
```
Comment (missionliao) [inline simulation/app.go]: Suggestion: maybe we could panic here.

File: simulation/app.go [hunk 6]
```diff
@@ -239,6 +240,7 @@ func (a *simApp) BlockDelivered(
 		Type:    blockTimestamp,
 		Payload: jsonPayload,
 	}
+	// #nosec G104
```
Comment (missionliao) [inline simulation/app.go]: Suggestion: maybe we could panic here.

File: simulation/config/config.go [hunk 7]
```diff
@@ -157,7 +157,7 @@ func GenerateDefault(path string) error {
 
 // Read reads the config from a file.
 func Read(path string) (*Config, error) {
-	f, err := os.Open(path)
+	f, err := os.Open(path) // #nosec G304
```
Comment (missionliao) [inline simulation/config/config.go]: Hmm, I'm not sure what's the best fix for this rule if what we provide is a library...

**Comment by jimmyhu-dexon** [comment]
I prefer not to add `panic` in the code where fullnode used for now.

File: simulation/config/config.go (hunk 7 above)
Comment (jimmyhu-dexon) [reply simulation/config/config.go]: This is intended. We want to read from a file as user input.

**Pull request merged**